pkg/authtoken/authtoken_test.go

```diff
@@ -0,0 +1,349 @@
+		t.Run(tt.name, func(t *testing.T) {
+			if got := ToCAPEMCertificateRawBytes(tt.args.certRawBytes); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("ToCAPEMCertificateRawBytes() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestNewDefaultTokenVerifier(t *testing.T) {
+	type args struct {
+		options TokenVerifierOptions
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    TokenVerifier
+		wantErr bool
+	}{
+		{
+			name: "should return an error",
+			args: args{
+				options: options,
+			},
+			want:    TokenVerifier{},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := NewDefaultTokenVerifier(tt.args.options)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("NewDefaultTokenVerifier() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewDefaultTokenVerifier() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_newTokenVerifier(t *testing.T) {
+	type args struct {
+		client  http.Client
+		options TokenVerifierOptions
+	}
+	tests := []struct {
+		name string
+		args args
+		want TokenVerifier
+	}{
+		{
+			name: "should return a TokenVerifier",
+			args: args{
+				client:  clientWithCACert,
+				options: options,
+			},
+			want: TokenVerifier{
+				client:  clientWithCACert,
+				options: options,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := newTokenVerifier(tt.args.client, tt.args.options); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("newTokenVerifier() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestTokenVerifier_Enabled(t *testing.T) {
+	type fields struct {
+		client  http.Client
+		options TokenVerifierOptions
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   bool
+	}{
+		{
+			name: "should return true",
+			fields: fields{
+				client:  clientWithCACert,
+				options: options,
+			},
+			want: true,
+		},
+		{
+			name: "should return false",
+			fields: fields{
+				client: clientWithCACert,
+				options: TokenVerifierOptions{
+					Enabled: false,
+					URL: url.URL{
+						Scheme: "https",
+						Host:   "localhost:8080",
+					},
+					CACertEnabled: true,
+					CACertRaw: CAPEMCertificateRawBytes{
+						a, b, c, d,
+					},
+				},
+			},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tr := &TokenVerifier{
+				client:  tt.fields.client,
+				options: tt.fields.options,
+			}
+			if got := tr.Enabled(); got != tt.want {
+				t.Errorf("TokenVerifier.Enabled() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestTokenVerifier_GetAuthenticationToken(t *testing.T) {
+	type fields struct {
+		client  http.Client
+		options TokenVerifierOptions
+	}
+	type args struct {
+		r *http.Request
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   string
+	}{
+		{
+			name: "should return token",
```

Comment:
See https://github.com/bf2fc6cc711aee1a0c2a/observability-remote-write-proxy/pull/23/files#r1165345194. Some of the test names should be more descriptive. Instead of telling what it returns it should explain in the form "when/if this happens this is the outcome" or similarly